enumerables.rb

```diff
@@ -0,0 +1,75 @@
+module Enumerable
+  def my_each
+    length.times { |value| yield(self[value]) }
+    self
+  end
+
+  def my_each_with_index
+    length.times { |value| yield(self[value], value) }
+    self
+  end
+
+  def my_select
+    array = []
+    my_each { |num| array << num if yield(num) }
+    array
+  end
+
+  def my_all?
+    result = true
+    my_each { |i| break result = false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each { |i| break result = true if yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each { |i| break result = false if yield(i) }
+    result
+  end
+
+  def my_count(arg)
+    counter = 0
+    my_each { |i| counter += 1 if arg == i }
+    counter
+  end
+
+  def my_map(proc = nil)
+    array = []
+    block_given? ? my_each { |i| array << yield(i) } : my_each { |i| array << proc.call(i) }
+    array
+  end
+
+  def my_inject(accumulator = 0)
+    my_each { |i| accumulator = yield(accumulator, i) }
+    accumulator
+  end
```

Comment:
- [ ] `my_inject` is not working as the original `inject` method, also is not passing all test cases.

### `inject`` is an Enumerable that has these characteristics:
- Returns a `LocalJumpError` when no block is given and no symbol is passed as an argument.
```
[1, 2, 3].inject
# LocalJumpError (no block given)
```

- it does not modify/change the original array/range/hash.

- it can accept an initial value and/or a symbol as an argument.
```
[1, 2, 3].inject(2, :*)
# 12
```

[Enumerable#inject](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-inject)